packages/preview-service/src/clients/knex.ts

```diff
@@ -1,29 +1,74 @@
 import { knexLogger as logger } from '@/observability/logging.js'
 import { getPostgresConnectionString, getPostgresMaxConnections } from '@/utils/env.js'
-import * as knex from 'knex'
-import { get } from 'lodash-es'
-
-// CJS interop (types are off)
-const knexBuilder = (get(knex, 'default') ||
-  get(knex, 'knex')) as unknown as typeof knex.knex
-
-export const db = knexBuilder({
-  client: 'pg',
-  connection: {
-    // eslint-disable-next-line camelcase
-    application_name: 'speckle_preview_service',
-    connectionString: getPostgresConnectionString()
-  },
-  pool: {
-    min: 0,
-    max: getPostgresMaxConnections(),
-    acquireTimeoutMillis: 16000, //allows for 3x creation attempts plus idle time between attempts
-    createTimeoutMillis: 5000
-  },
-  log: {
-    warn: (message) => logger.warn(message),
-    error: (message) => logger.error(message),
-    debug: (message) => logger.debug(message)
+import Environment from '@speckle/shared/dist/commonjs/environment/index.js'
+import {
+  loadMultiRegionsConfig,
+  configureKnexClient
+} from '@speckle/shared/dist/commonjs/environment/multiRegionConfig.js'
+import { Knex } from 'knex'
+
+const { FF_WORKSPACES_MULTI_REGION_ENABLED } = Environment.getFeatureFlags()
+
+type ConfiguredKnexClient = ReturnType<typeof configureKnexClient>
+export type DbClients = Record<'main', ConfiguredKnexClient> &
+  Record<string, ConfiguredKnexClient>
+let dbClients: DbClients
+
+const isDevEnv = process.env.NODE_ENV === 'development'
+
+export const getDbClients = async () => {
```

Comment:
this is already better than it was, but in the future we can probably even extract a lot of this DB querying/"repository" logic that is common between these BG services into shared as well